src/lib/dlt_user.c

```diff
@@ -485,7 +488,8 @@ DltReturnValue dlt_init(void)
 
     /* Initialize common part of dlt_init()/dlt_init_file() */
     if (dlt_init_common() == DLT_RETURN_ERROR) {
-        dlt_user_init_state = INIT_UNITIALIZED;
+        dlt_user_init_state = INIT_DONE;
```

Comment:
I would prefer to leave the state set to INIT_IN_PROGRESS and check for this state in dlt_free() as well.

It is quite confusing to set state to INIT_DONE in failure case.

Reply:
Yes, INIT_ERROR would be also a name for this state.

Reply:
added INIT_ERROR state